src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1248,6 +1265,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Debug.Assert(_moduleBeingBuiltOpt Is Nothing OrElse _moduleBeingBuiltOpt.AllowOmissionOfConditionalCalls)
 
+            Dim lambdaDebugInfoBuilder = ArrayBuilder(Of LambdaDebugInfo).GetInstance()
```

Comment:
Consider moving closer to Rewriter.LowerBodyOrInitializer call, so the ArrayBuilders are only allocated when necessary and to ensure the ArrayBuilders are empty.